### PR TITLE
Add GenerateAlphaNumericID for test-proxy to recording package

### PR DESCRIPTION
**Pull request by tadelesh**
1. Add GenerateAlphaNumericID function.
2. Store random seed in each recording file.
3. This change is to move the similar function under `sdk/resourcemanger/internal` as it should be part of our recording system. 

Part of #19984

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
@tadelesh can you please try running the recorded tests for `azblob` with this change to ensure that nothing is broken?

File: sdk/internal/recording/recording.go [hunk 1]
```diff
@@ -814,7 +801,76 @@ func Stop(t *testing.T, options *RecordingOptions) error {
...
+		testStruct.recordingSeed = seed
+		testStruct.recordingRandomSource = source
+		testSuite.Store(t.Name(), testStruct)
+	}
+
+	return source
+}
+
+// GenerateAlphaNumericID will generate a recorded random alpha numeric id.
+// When playback mode, a new random seed will be used, otherwise, the random seed is stable and will be stored in recording file.
```
Comment (jhendrixMSFT) [inline sdk/internal/recording/recording.go:836]: I think this comment is wrong and should be the same as `Recording.GenerateAlphaNumericID()` yes?

File: sdk/internal/recording/recording.go [hunk 2]
```diff
@@ -496,22 +466,27 @@ func init() {
...
 
 type recordedTest struct {
-	recordingId string
-	liveOnly    bool
-	variables   map[string]interface{}
+	recordingId           string
+	liveOnly              bool
+	variables             map[string]interface{}
+	recordingSeed         int64
+	recordingRandomSource rand.Source
```
Comment (jhendrixMSFT) [inline sdk/internal/recording/recording.go:489]: Can we shorten the name of this field a bit?  Maybe `recordingRndSrc`?

File: sdk/internal/recording/recording.go [hunk 3]
```diff
@@ -778,6 +758,12 @@ func Stop(t *testing.T, options *RecordingOptions) error {
 			// test should only be run live, don't want to generate recording
 			return nil
 		}
+		if testStruct.recordingSeed != 0 {
```
Comment (jhendrixMSFT) [inline sdk/internal/recording/recording.go:761]: How did this used to work, or was it broken?

File: sdk/internal/recording/recording.go [hunk 4]
```diff
@@ -496,22 +466,27 @@ func init() {
...
-	IDHeader          = "x-recording-id"
-	ModeHeader        = "x-recording-mode"
-	UpstreamURIHeader = "x-recording-upstream-base-uri"
+	RecordingMode                   = "record"
+	PlaybackMode                    = "playback"
+	LiveMode                        = "live"
+	IDHeader                        = "x-recording-id"
+	ModeHeader                      = "x-recording-mode"
+	UpstreamURIHeader               = "x-recording-upstream-base-uri"
+	recordingRandomSeedVariableName = "recordingRandomSeed"
```
Comment (jhendrixMSFT) [inline sdk/internal/recording/recording.go:481]: Can we shorten this name?

File: sdk/internal/recording/recording.go (hunk 3 above)
Comment (tadelesh) [reply sdk/internal/recording/recording.go:761]: For the test-proxy testing framework, we don't have utility to support generating random string. So, we need to store the random seed in the option's variable and stop with this option to store the random seed. For now, we add such feature, so we need to do the seed store for user.

**Comment by tadelesh** [comment]
> @tadelesh can you please try running the recorded tests for `azblob` with this change to ensure that nothing is broken?

I've tested `azblob` and all tests passed.

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**